### ISSUE TITLE
Can't return enum from a method

### ISSUE THREAD

**Issue by pooroligarch** [godotengine/godot-cpp#753]
When returning an enum value from a method compilation fails with a message like this:
```
/home/tpotancok/Godot/godot-cpp/include/godot_cpp/core/method_bind.hpp: In instantiation of ‘GDNativeVariantType godot::MethodBindTR<R, P>::gen_argument_type(int) const [with R = ExampleRef::BaseColor; P = {}]’:
/home/tpotancok/Godot/godot-cpp/include/godot_cpp/core/method_bind.hpp:430:30:   required from here
/home/tpotancok/Godot/godot-cpp/include/godot_cpp/core/method_bind.hpp:434:48: error: cannot convert ‘const godot::Variant::Type’ to ‘GDNativeVariantType’ in return
  434 |                         return GetTypeInfo<R>::VARIANT_TYPE;
      |                                                ^~~~~~~~~~~~
```
Registering enums works just fine. I can work around this by returning an integer instead but using the enum directly is cleaner and probably faster.

Edit: I also get this when taking an enum as an argument, so they're completely unusable outside of internal data structures.

Reproduction project: [enum-return.tar.gz](https://github.com/godotengine/godot-cpp/files/8672432/enum-return.tar.gz)

**Comment by pooroligarch** [comment]
@Calinou I found this bug on `master` branch, not sure if it also affects 3.x

**Comment by pooroligarch** [comment]
Fixed by #757.
